### ISSUE TITLE
Implement desktop view

### ISSUE THREAD

**Issue by ferndot** [apc-io/apc-firefox-os#20]
Since we are porting Firefox OS to a desktop device, it would probably be good to customize the interface to suit larger screens and resolutions.

We should also implement a model system for apps (maybe reading things from meta tags).

I hope it was okay for me to create this issue.  If it wasn't I will remove it.

**Comment by moskovich** [comment]
Definitely. Feel free to create issues that you think we need to discuss.

```
In this particular case, our plan was to do as few changes as possible and push these changes back to Mozilla. Recently I think there will be more than a few changes that might not be easy for them to merge quickly, so yes, I believe we will need something like this.   

I will let Le Chau comment more on the details with her plan. 

—
```

Sean

**Comment by ferndot** [comment]
This is being worked on in https://github.com/apc-io/apc_b2g_gaia/tree/desktop-ui.

**Comment by woutervddn** [comment]
I'm wondering what the best approach for this is.. I only skimmed through the code, but it seems like it will enable desktop view by default when the screen resolution is high enough. This makes it hard to commit upstream.

I think it might be cleaner to add an "enable desktop mode" feature to the settings menu. Code compatible with that structure would be more easy to commit upstream. This also ensures that a desktop mode can be enabled and disabled on a tablet with a keyboard attached. Gecko could make a trigger that whenever a keyboard is attached, the desktop mode turns on. (unless forced by the settings menu)

**Comment by ferndot** [comment]
That sounds like a good idea.

:+1:
